MAVProxy/modules/mavproxy_swarm.py

```diff
@@ -0,0 +1,953 @@
+        self.killAllTimer = None
+
+    def kill(self, event):
+        '''disarm force (kill) the drone'''
+        if self.doKill.GetLabel() == "KILL":
+            self.doKill.SetLabel("KILL Sure?")
+            # start a 1 sec timer
+            self.killTimer = wx.Timer(self)
+            self.Bind(wx.EVT_TIMER, self.killTimeout, self.killTimer)
+            self.killTimer.Start(1000)
+        else:
+            self.state.child_pipe.send(("kill", self.sysid, self.compid))
+
+    def killall(self, event):
+        '''disarm force (kill) the whole swarm'''
+        if self.doKillAll.GetLabel() == "ALL":
+            self.doKillAll.SetLabel("ALL Sure?")
+            # start a 1 sec timer
+            self.killAllTimer = wx.Timer(self)
+            self.Bind(wx.EVT_TIMER, self.killAllTimeout, self.killAllTimer)
+            self.killAllTimer.Start(1000)
+        else:
+            self.state.child_pipe.send(("kill", self.sysid, self.compid))
+            if len(self.listFollowers) > 0:
+                for (sysidFollow, compidFollow, vehtype) in self.listFollowers:
+                    self.state.child_pipe.send(
+                        ("kill", sysidFollow, compidFollow))
+
+    def follow(self, event):
+        '''switch to mode follow'''
+        self.state.child_pipe.send(("FOLLOW", self.sysid, self.compid))
+
+    def followAll(self, event):
+        '''Mode FOLLOW for all followers'''
+        if len(self.listFollowers) > 0:
+            for (sysidFollow, compidFollow, vehtype) in self.listFollowers:
+                self.state.child_pipe.send(
+                    ("FOLLOW", sysidFollow, compidFollow))
+
+    def auto(self, event):
+        '''switch to mode AUTO'''
+        self.state.child_pipe.send(("AUTO", self.sysid, self.compid))
+
+    def updateOffset(self, param, value):
+        '''get offset param'''
+        if self.isLeader:
+            return
+        if param == 'FOLL_OFS_X':
+            self.offsetValues[0] = value
+        elif param == 'FOLL_OFS_Y':
+            self.offsetValues[1] = value
+        elif param == 'FOLL_OFS_Z':
+            self.offsetValues[2] = value
+        else:
+            return
+        # refresh static text
+        if self.offsetValues[0] != None and self.offsetValues[1] != None and self.offsetValues[2] != None:
+            self.offsets.SetLabel("Offset: {0}m,{1}m,{2}m".format(
+                self.offsetValues[0], self.offsetValues[1], self.offsetValues[2]))
+
+
+class swarmFrame(wx.Frame):
+    '''
+    The main frame of the UI. Runs in a subprocess, so need a pipe to communicate with
+    '''
+
+    def __init__(self, state, title, params, takeoffalt):
+        self.state = state
+        wx.Frame.__init__(self, None, title=title, size=(
+            1400, 500), style=wx.DEFAULT_FRAME_STYLE)
+        self.panel = scrolled.ScrolledPanel(self)
+
+        # Icon
+        try:
+            self.SetIcon(icon.SimpleIcon("CONSOLE").get_ico())
+        except Exception:
```

Comment:
`Exception` should never be caught; surely there's some tighter exception you can catch?